src/main/java/cz/cvut/kbss/analysis/service/FaultTreeRepositoryService.java

```diff
@@ -382,4 +381,23 @@ protected boolean removeVisited(Behavior b){
         Set<Behavior> visited = visitedBehaviors.get();
         return visited.remove(b);
     }
+
+    private void setFaultEventTypes(Behavior behaviorChild, FaultEvent fEvent) {
+        if (behaviorChild.getChildBehaviors().isEmpty()
+                && behaviorChild.getRequiredBehaviors().isEmpty()
+                && functionRepositoryService.getImpairingBehaviors(behaviorChild.getUri()).isEmpty())
+        {
+            fEvent.setEventType(EventType.BASIC);
+            fEvent.setGateType(GateType.UNUSED);
+            fEvent.setProbability(1.);
+        } else {
+            fEvent.setEventType(EventType.INTERMEDIATE);
+            fEvent.setGateType(GateType.OR);
+        }
+    }
+
+    private boolean isFailureModeCause(Behavior behavior) {
+        return behavior instanceof FailureMode
+                && ((FailureMode) behavior).getFailureModeType() == FailureModeType.FailureModeCause;
+    }
```

Comment:
This function seams generic enough to be added to the Behavior entity.

Reply:
Ok